src/EA.Weee.DataAccess.Tests.DataAccess/StoredProcedure/SpgSchemeObligationCsvDataTests.cs

```diff
@@ -62,6 +67,94 @@ public async Task Execute_HappyPath_ReturnsProducerEeeForPreviousYear()
             }
         }
 
+        [Fact]
+        public async Task Execute_ReturnsProducerEeeData_For_All_Categories()
+        {
+            using (DatabaseWrapper db = new DatabaseWrapper())
+            {
+                //Arrange
+                string prn1 = "PRN123";
+                string prn2 = "PRN456";
+                int complianceYear = DateTime.Now.Year;
+                string approvalNumber = "WEE/TE0000ST/SCH";
+                string b2bObligationType = EnumHelper.GetDisplayName(ObligationType.B2B);
+                string b2cObligationType = EnumHelper.GetDisplayName(ObligationType.B2C);
+
+                ModelHelper helper = new ModelHelper(db.Model);
+                var scheme = helper.CreateScheme();
+                scheme.ApprovalNumber = approvalNumber;
+
+                var categories = EnumHelper.GetValues(typeof(WeeeCategory));
+                var maxCategoryId = categories.Max(x => x.Key);
+                int quarter1Tonnage = 0;
+                int quarter2Tonnage = 0;
+
+                //Previous Year Data
+                var memberUpload1 = helper.CreateSubmittedMemberUpload(scheme);
+                memberUpload1.ComplianceYear = complianceYear - 1;
+                var prod1 = helper.CreateProducerAsCompany(memberUpload1, prn1, b2cObligationType);
```

Comment:
the names in this test are pretty unhelpful (worse than in previous tests, though it's following a similar pattern). It would be a lot clearer what's going on if `memberUpload1` were `memberUploadPreviousYear`, then `prod1` was `prodPrn1PreviousYear`, and so on, so `prod3` and `prod4` end up with much clearer names, and it's then more obvious that both `prod4` and `prod2` should be excluded because they both related to `prn2`.

Reply:
hope that makes sense -- I'm fine with `prn1` and `prn2`, it's just that when all the other variables relate to these two in some way, it would be a lot easier if the connection was in the name, as well as what's different about them.